fdbserver/storageserver.actor.cpp

```diff
@@ -7778,6 +7714,119 @@ ACTOR Future<Void> replaceTSSInterface(StorageServer* self, StorageServerInterfa
 	return Void();
 }
 
+ACTOR Future<Void> storageInterfaceRegistration(StorageServer* self,
+                                                StorageServerInterface ssi,
+                                                Future<bool> interfaceAcceptingRequests) {
```

Comment:
IMO the signature of this function should be something like:
```
ACTOR static Future<Void> storageInterfaceRegistration(StorageServer *self, StorageServerInterface ssi, Optional<Future<Void>> readyToAcceptRequests = {});
```
Then when calling this function in the `storageServer` overload for existing storage servers, you don't need to create a new temporary `Promise<bool>` object. The `storageInterfaceRegistration` function can immediately register with `StorageServerInterface::acceptingRequests` set to false. Also, the `registerInterfaceAcceptingRequests` and `interfacedRegistered` fields can then be collapsed into a single `Promise<Void>` field.

Reply:
After some offline discussion, it looks like we still need two fields, but their types can be `Future<Void>` and `Promise<Void>`.